src/js/utils/DOM.js

```diff
@@ -67,25 +67,13 @@ export const containsFocus = (node) => {
 // Check if the element.tagName is an input, select or textarea
 export const isFocusable = (element) => {
   const tagName = element.tagName.toLowerCase();
-  return tagName === 'input' || tagName === 'select' || tagName === 'textarea';
-};
-
-// Get the first element that can receive focus
-export const getFirstFocusableDescendant = (element) => {
-  const children = element.getElementsByTagName('*');
-  for (let i = 0; i < children.length; i += 1) {
-    const child = children[i];
-    if (isFocusable(child)) {
-      return child;
-    }
-  }
-  return undefined;
-};
-
-export const shouldKeepFocus = () => {
-  const element = document.activeElement;
-  if (isFocusable(element)) return true;
-  return !!getFirstFocusableDescendant(element);
+  return (
+    tagName === 'input' ||
+    tagName === 'select' ||
+    tagName === 'textarea' ||
+    tagName === 'button' ||
+    element.tabIndex >= 0
```

Comment:
Why exclude -1? -1 takes it out of the tab order but you can still get focus on it a couple different ways.

Reply:
Including -1 results in too many elements being marked as focusable. For example, with the Select component 9 elements will be marked as focusable including Boxes, Icons, Svgs, and others because they have a tabIndex of -1.